lib/commons/builder/wikidata_queries.rb

```diff
@@ -31,213 +38,39 @@ def query_legislative(position_item_id:, house_item_id:, term_item_id: nil, star
     unless !!term_item_id ^ !!(start_date and end_date)
       raise 'You must specify either a term item or a start and end date (and not both)'
     end
-    <<~SPARQL
-      SELECT ?statement
-             ?item #{lang_select}
-             ?party #{lang_select('party_name')}
-             ?district #{lang_select('district_name')}
-             ?role #{lang_select('role')}
-             ?role_superclass #{lang_select('role_superclass')}
-             ?start ?end ?facebook
-             ?org #{lang_select('org')} ?org_jurisdiction ?org_seat_count
-      WHERE {
-        BIND(wd:#{position_item_id} as ?role) .
-        BIND(wd:#{house_item_id} as ?org) .
-        #{lang_options('org', '?org')}
-        OPTIONAL {
-          ?org wdt:P1001 ?org_jurisdiction
-        }
-        OPTIONAL {
-          ?org wdt:P1342 ?org_seat_count
-        }
-        ?item p:P39 ?statement .
-        #{lang_options}
-        ?statement ps:P39 ?role .
-        #{lang_options('role', '?role')}
-        OPTIONAL {
-          ?role wdt:P279 ?role_superclass .
-          ?role_superclass wdt:P279+ wd:Q4175034
-          #{lang_options('role_superclass', '?role_superclass')}
-        }
-        #{term_condition(term_item_id)}
-        OPTIONAL { ?statement pq:P580 ?start }
-        OPTIONAL { ?statement pq:P582 ?end }
-        OPTIONAL {
-          ?statement pq:P768 ?district.
-          #{lang_options('district_name', '?district')}
-        }
-        OPTIONAL {
-          ?statement pq:P4100 ?party.
-          #{lang_options('party_name', '?party')}
-        }
-        OPTIONAL { ?item wdt:P2013 ?facebook }
-        #{date_condition(start_date, end_date)}
-      } ORDER BY ?item ?role #{term_item_id ? '?term ' : ''}?start ?end
-  SPARQL
+    query = templated_query('legislative')
+    query.result binding
```

Comment:
I'm a bit uneasy about passing `binding` to a template, rather than explicitly specifying what objects should be available in the template, but this does seem to be common practice for using ERB (which is much less safe than Django templates anyway, which I guess is what I'm comparing it to).